sql/core/src/test/resources/sql-tests/inputs/pgSQL/boolean.sql

```diff
@@ -0,0 +1,265 @@
+--
+-- Portions Copyright (c) 1996-2019, PostgreSQL Global Development Group
+--
+--
+-- BOOLEAN
+-- https://github.com/postgres/postgres/blob/REL_12_BETA1/src/test/regress/sql/boolean.sql
+
+--
+-- sanity check - if this fails go insane!
+--
+SELECT 1 AS one;
+
+
+-- ******************testing built-in type bool********************
+
+-- check bool input syntax
+
+SELECT true AS true;
+
+SELECT false AS false;
+
+SELECT cast('t' as boolean) AS true;
```

Comment:
We use `cast('t' as boolean)` to workaround `bool 't'`. Do we need to support `bool 't'`?

Reply:
Check the other SQL engines?

Reply:
It seems it's PostgreSQL specific.

Reply:
Instead of creating built-in UDF functions, can we create these UDFs in our SQLQueryTestSuite?

Reply:
How about using our built-in UDF `boolean`: https://github.com/apache/spark/blob/d169b0aac369d373968a6c66ee43440a2ad751a5/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/FunctionRegistry.scala#L524

It seems we need add support create boolean typed Literal expression if we want support `bool 't'`:
https://github.com/apache/spark/blob/2da5b21834a4633af36c98297a5368c66b8d3191/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala#L1605-L1637

Reply:
That works for me.